tests/phpunit/tests/post/getPageByTitle.php

```diff
@@ -0,0 +1,242 @@
+<?php
+
+/**
+ * @group post
+ * @covers ::get_page_by_title
+ */
+class Tests_Post_GetPageByTitle extends WP_UnitTestCase {
+	/**
+	 * @ticket 36905
+	 */
+	public function test_get_page_by_title_priority() {
+		$attachment = self::factory()->post->create_and_get(
+			array(
+				'post_title' => 'some-other-page',
+				'post_type'  => 'attachment',
+			)
+		);
+		$page       = self::factory()->post->create_and_get(
+			array(
+				'post_title' => 'some-page',
+				'post_type'  => 'page',
+			)
+		);
+
+		// get_page_by_title() should return a post of the requested type before returning an attachment.
+		$this->assertEquals( $page, get_page_by_title( 'some-page' ) );
```

Comment:
For tests that have multiple assertions, please add a message to each assertion so it is clear which fails if there is a failure.

Reply:
Fixed.